component/main.jsonnet

```diff
@@ -1,16 +1,34 @@
-local com = import 'lib/commodore.libjsonnet';
+local acme = import 'acme.libsonnet';
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
 
 local params = inv.parameters.openshift4_ingress;
+local usesAcme(name) = !std.objectHas(params.ingressControllers[name], 'defaultCertificate');
+local anyControllerUsesAcme = std.foldl(
+  function(x, field) x || usesAcme(field),
+  std.objectFields(params.ingressControllers),
+  false,
+);
 
 {
-  [name]: kube._Object('operator.openshift.io/v1', 'IngressController', name) {
-    metadata+: {
-      namespace: params.namespace,
-    },
-    spec: params.ingressControllers[name],
-  }
+  local acmeCertName = 'acme-wildcard-' + name,
+  [name]:
+    [kube._Object('operator.openshift.io/v1', 'IngressController', name) {
+      metadata+: {
+        namespace: params.namespace + '-operator',
+      },
+      spec: params.ingressControllers[name] {
```

Comment:
And now, we can change the order of objects and eliminate the need for a conditional. This would be possible by setting the static defaults in the first object. The values from `params.ingressControllers[name]` will then override those defaults.

Reply:
Nice, I like this 👍